src/mscorlib/src/System/ArraySegment.cs

```diff
@@ -106,12 +110,15 @@ public int Count
             }
         }
 
-        public Enumerator GetEnumerator()
+        public T this[int index]
         {
-            if (_array == null)
-                ThrowHelper.ThrowInvalidOperationException(ExceptionResource.InvalidOperation_NullArray);
-            Contract.EndContractBlock();
+            get { return _array[_offset + index]; }
```

Comment:
@jkotas Well, a quick GitHub search through Kestrel shows a lot of usages... https://github.com/aspnet/KestrelHttpServer/search?utf8=%E2%9C%93&q=ArraySegment. [Here](https://github.com/aspnet/KestrelHttpServer/blob/4a1db745687098cc4e8273f03b2c8a2754587f1a/src/Microsoft.AspNetCore.Server.Kestrel/Internal/Networking/UvWriteReq.cs#L137) is one place where this API could make code more readable.